pkg/reconciler/build/build.go

```diff
@@ -107,29 +122,205 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 	}
 
 	// Get the Build resource with this namespace/name
-	if _, err := c.buildsLister.Builds(namespace).Get(name); errors.IsNotFound(err) {
+	// TODO(jasonhall): Use a lister?
+	orig, err := c.buildclientset.BuildV1alpha1().Builds(namespace).Get(name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
 		// The Build resource may no longer exist, in which case we stop processing.
 		logger.Errorf("build %q in work queue no longer exists", key)
 		return nil
 	} else if err != nil {
 		return err
 	}
+	// Don't modify the informer's copy.
+	b := orig.DeepCopy()
 
-	// TODO(jasonhall): adopt the standard reconcile pattern.
-	// For Build this looks something like:
-	// podName := names.Pod(build)
-	// pod, err := c.podLister.Pods(build.Namespace).Get(podName)
-	// if IsNotFound(err) {
-	// 	desired := resources.MakePod(build)
-	// 	pod, err = c.kubeclientset.V1().Pods(build.Namespace).Create(desired)
-	// 	if err != nil {
-	// 		return err
-	// 	}
-	// } else if err != nil {
-	// 	return err
-	// }
-	//
-	// // Update build.Status based on pod.Status
+	// Reconcile this copy of the build and then write back any status
+	// updates regardless of whether the reconciliation errored out.
+	err = c.reconcile(ctx, b)
+	if equality.Semantic.DeepEqual(orig.Status, b.Status) {
+		// If we didn't change anything then don't call updateStatus.
+		// This is important because the copy we loaded from the
+		// informer's cache may be stale and we don't want to overwrite
+		// a prior update to status with this stale state.
+	} else if err := c.updateStatus(b); err != nil {
+		logger.Warn("Failed to update build status", zap.Error(err))
+		return err
+	}
+	return err
+}
+
+func (c *Reconciler) reconcile(ctx context.Context, b *v1alpha1.Build) error {
+	logger := logging.FromContext(ctx)
+
+	// If the build doesn't define a pod name, it means it's new. We should
+	// apply any template and create its pod.
+	if b.Status.Cluster == nil || b.Status.Cluster.PodName == "" {
```

Comment:
First thing to check would be build status whether its completed or not. If build is completed, ` b.Status.Cluster.PodName` will be set but `GetPod()` will fail with not found error.